libbeat/reader/syslog/util_test.go

```diff
@@ -255,3 +255,48 @@ func TestAppendStringField(t *testing.T) {
 		})
 	}
 }
+
+func TestJoinStr(t *testing.T) {
+	tests := map[string]struct {
+		InA   string
+		InB   string
+		InSep string
+		Want  string
```

Comment:
When I read the function I was wondering why not use `strings.Join` which is why I asked. I'd say that the test is of behaviour rather than structure, so a reference implementation to compare against is more informative, but I'm not fussed by it.

Reply:
Ah I see. Yeah this function is to get around that I don't want a space added to the result if one of the parameters (A or B) is an empty string. `strings.Join` isn't capable of doing that, hence why it isn't used.

Reply:
Thanks for explaining.